drivers/w3s.go

```diff
@@ -336,19 +337,41 @@ func ipfsCarPack(ctx context.Context, filePath string) (string, string, error) {
 }
 
 // w3StoreCar uses external binary `w3` to store a CAR file in web3.storage.
-func w3StoreCar(ctx context.Context, carPath string) (string, error) {
-	out, err := exec.CommandContext(ctx, "w3", "can", "store", "add", carPath).Output()
+func w3StoreCar(ctx context.Context, proof, carPath string) (string, error) {
+	out, err := runWithCredentials(exec.CommandContext(ctx, "livepeer-w3", "can", "store", "add", carPath), proof)
 	if err != nil {
 		return "", err
 	}
 	return strings.TrimSpace(string(out)), nil
 }
 
 // w3StoreCar uses external binary `w3` to bind and publish multiple CARs.
-func w3UploadCar(ctx context.Context, rootCid string, carCids []string) error {
+func w3UploadCar(ctx context.Context, proof, rootCid string, carCids []string) error {
 	args := []string{"can", "upload", "add"}
 	args = append(args, rootCid)
 	args = append(args, carCids...)
-	_, err := exec.CommandContext(ctx, "w3", args...).Output()
+	_, err := runWithCredentials(exec.CommandContext(ctx, "livepeer-w3", args...), proof)
 	return err
 }
+
+func runWithCredentials(cmd *exec.Cmd, proof string) ([]byte, error) {
+	if proof == "" {
+		return nil, fmt.Errorf("UCAN proof not found")
+	}
+	base64Proof, err := base64UrlToBase64(proof)
+	if err != nil {
+		return nil, fmt.Errorf("invalid UCAN proof format: %s", err)
+	}
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, fmt.Sprintf("W3_PRINCIPAL_KEY='%s'", W3sUcanKey))
```

Comment:
I wonder if there is an opp to simplify how this key is configured. I see that right now the flow is:

- Pass value in a flag for a catalyst-api
- Set the package var `W3sUcanKey` with the value
- Read `W3sUcanKey` and set it as an env var when calling the livepeer-w3 script

An alternative might be:

- Set `W3_PRINCIPAL_KEY` in the environment that catalyst-api is running in i.e. in the Docker container
- livepeer-w3 just reads the `W3_PRINCIPAL_KEY` env var value directly without any additional value passing which reduces the # of code paths that have to be aware of this value

I see we've already merged the relevant catalyst-api PRs so maybe consider whether this might be a worthwhile simplification separately.

Reply:
My initial thinking was that not storing the private key in the environment variable is more secure because even if you access the container, you still cannot access it.

However, now I think you're right because anyway we pass the key in the env variable to catalyst. So, maybe let's get rid of the Catalyst flag and just use the private key passed in the `W3_PRINCIPAL_KEY`. Thanks for this comment @yondonfu